compiler-core/src/config.rs

```diff
@@ -62,25 +63,72 @@ impl<'de> Deserialize<'de> for SpdxLicense {
     }
 }
 
+impl Serialize for SpdxLicense {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.licence)
+    }
+}
+
 impl AsRef<str> for SpdxLicense {
     fn as_ref(&self) -> &str {
         self.licence.as_str()
     }
 }
 
-#[derive(Deserialize, Debug, PartialEq, Clone)]
+#[derive(Debug, PartialEq, Clone)]
+pub struct GleamVersion {
+    pubgrub: pubgrub::range::Range<Version>,
+    hex: version::Range,
+}
+
+impl GleamVersion {
+    pub fn from_pubgrub(range: pubgrub::range::Range<Version>) -> Self {
+        Self {
+            hex: version::Range::new(range.to_string()),
+            pubgrub: range,
+        }
+    }
+
+    pub fn as_pubgrub(&self) -> pubgrub::range::Range<Version> {
+        self.pubgrub.clone()
+    }
+
+    pub fn new(spec: String) -> Result<GleamVersion> {
+        let hex = version::Range::new(spec.to_string());
+        let pubgrub = hex.to_pubgrub().map_err(|e| Error::InvalidVersionFormat {
+            input: spec,
+            error: e.to_string(),
+        })?;
+
+        Ok(Self { pubgrub, hex })
+    }
+
+    pub fn lowest_version(&self) -> Option<Version> {
+        self.pubgrub.lowest_version()
+    }
+
+    pub fn hex(&self) -> version::Range {
+        self.hex.clone()
```

Comment:
```suggestion
    pub fn hex(&self) -> &version::Range {
        &self.hex
```

Reply:
I get this:

```
cannot move out of `self.hex` which is behind a shared reference
move occurs because `self.hex` has type `hexpm::version::Range`, which does not implement the `Copy` trait
```